metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -1123,115 +993,24 @@ def _get_upstream_lineage_info_projection(
-                basename = db_row_key["basename"]
-                upstream = f"{db_row_key['schemaname']}.{db_row_key['basename']}"
-
-                view_downstream_query = sql.text(
-                    dedent(
-                        """
-                    select name,schemaname
-                    from vs_projections
-                    where basename='%(basename)s'
-                    """
-                        % {"basename": basename}
-                    )
-                )
-                for db_row_value in engine.execute(view_downstream_query):
-                    downstream = f"{db_row_value['schemaname']}.{db_row_value['name']}"
-                    projection_upstream: str = upstream
-                    projection_name: str = downstream
-                    self.projection_lineage_map[projection_name].append(
-                        # (<upstream_table_name>, <empty_json_list_of_upstream_table_columns>, <empty_json_list_of_downstream_view_columns>)
-                        (projection_upstream, "[]", "[]")
-                    )
-                    num_edges += 1
-
-        except Exception as error:
-            logger.warning(
-                "Extracting the upstream & downstream Projection lineage from Vertica failed %s",
-                error,
-            )
-
-        logger.info(
-            f"A total of {num_edges} Projection lineage edges found for {projection}."
-        )
-
-    def _get_owner_information(self, table: str, label: str) -> Optional[str]:
-        url = self.config.get_sql_alchemy_url()
-        engine = create_engine(url, **self.config.options)
-        if label == "table":
-            get_owner_query = sql.text(
-                dedent(
-                    """
-                    SELECT owner_name
-                    FROM v_catalog.tables
-                    WHERE table_name = '%(table)s'
-                    """
-                    % {"table": table}
-                )
-            )
-
-            for each in engine.execute(get_owner_query):
-                return each["owner_name"]
-        elif label == "view":
-            get_owner_query = sql.text(
-                dedent(
-                    """
-                    SELECT owner_name
-                    FROM v_catalog.views
-                    WHERE table_name =  '%(view)s'
-                    """
-                    % {"view": table}
-                )
-            )
-
-            for each in engine.execute(get_owner_query):
-                return each["owner_name"]
-
-        elif label == "projection":
-            get_owner_query = sql.text(
-                dedent(
-                    """
-                    SELECT owner_name
-                    FROM v_catalog.projections
-                    WHERE projection_name = '%(projection)s'
-                    """
-                    % {"projection": table}
-                )
+            columns = inspector.get_all_columns(table, schema)
```

Comment:
was there a reason to call this get_all_columns and not get_columns?

Reply:
get_columns is used by profiling and its query are table level. We had to create a new function with schema level query .

Reply:
Critically: `get_columns` should still do the right thing, and do it efficiently.

Reply:
It does not . Get_all_columns is duplicate of get_columns . The reason for duplication was that profiling also calls get_columns . If we used get_columns call get_all_coulmns . Profiling will also call it  . We wanted to keep profiling totally seperate from other features of plugin .  Please check get_all_columns here - https://github.com/vertica/vertica-sqlalchemy-dialect/blob/main/vertica_sqlalchemy_dialect/base.py .

Reply:
resolved , as we removed get_all_columns and used get_columns